### DISCUSSION TITLE
Bind a button/fill-in widget to a frame

### DISCUSSION THREAD

**Discussion by M4SSD35TRUCT10N** [fltk-rs/fltk-rs#854] [Q&A]
I'm coming from a different programming language and I'm used to "bind"/create a widget on a frame and then hide or show the whole frame. Now I'm using fltk-rs for a while (still learning Rust and FLTK-RS at the same (spare)time) and couldn't do that. Does anyone know how to do that?

Many thanks in advance.

**Comment by MoAlyousef** [comment]
Hi
An fltk Frame, can't have children. But you have several group types which can. A group type is any widget implementing the GroupExt trait.

```rust
use fltk::{prelude::*, *};

const SHOW_GUI: bool = true;

fn create_group() -> group::Group {
    let mut group = group::Group::default().size_of_parent();
    let mut btn = button::Button::default()
        .with_size(100, 30)
        .with_label("Click")
        .center_of_parent();
    group.end();
    group.hide();
    btn.set_callback(|b| b.parent().unwrap().hide());
    group
}

fn main() {
    let a = app::App::default();
    let mut win = window::Window::default().with_size(400, 300);
    if SHOW_GUI {
        let mut grp = create_group();
        grp.show();
    }
    win.end();
    win.show();
    a.run().unwrap();
}
```
